src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php

```diff
@@ -121,45 +122,14 @@ public function build()
                 $this->webspaces[] = $webspace;
 
                 $this->buildPortals($webspace);
-            } catch (\InvalidArgumentException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidUrlDefinitionException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidCustomUrlException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (WebspaceLocalizationNotUsedException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
+            } catch (\Exception $e) {
```

Comment:
We catch the exception and wrap it in the `InvalidWebspaceException`.
The `InvalidWebspaceException` contains a message with the path of the
webspace which could not be loaded. The previous exception will be shown
after the primary exception, so we don't lose any information.

On Tue, Apr 12, 2016 at 12:31:44AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php:
> 
>  > -                $this->logger->warning(
>  > -                    sprintf(
>  > -                        'Error: "%s" in "%s". The file has been skipped.',
>  > -                        $e->getMessage(),
>  > -                        $file->getRealPath()
>  > -                    )
>  > -                );
>  > -            } catch (WebspaceLocalizationNotUsedException $e) {
>  > -                $this->logger->warning(
>  > -                    sprintf(
>  > -                        'Error: "%s" in "%s". The file has been skipped.',
>  > -                        $e->getMessage(),
>  > -                        $file->getRealPath()
>  > -                    )
>  > -                );
>  > +            } catch (\Exception $e) {
> 
>    I am against catching the base exception class, since also errors not
>    related to webspaces could be swallowed here. Actually, why should we
>    catch an exception here at all? We are rethrowing an exception, making the
>    path to follow unnecessarily harder.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu/pull/2279#discussion_r59331968
>    2. https://github.com/sulu/sulu/pull/2279/files/488904213abfd5de2073dc30473f858099ce741a#r59331968
>    3. https://github.com/sulu/sulu/pull/2279/files/488904213abfd5de2073dc30473f858099ce741a#r59331968